core/src/test/java/org/apache/iceberg/deletes/TestRoaringPositionBitmap.java

```diff
@@ -0,0 +1,516 @@
+    bitmap.addRange(
+        position(1 /* bitmap */, 1 /* container */, 10L),
+        position(1 /* bitmap */, 1 /* container */, 500L));
+
+    // bitmap 1, container 2 (bitset)
+    bitmap.addRange(
+        position(1 /* bitmap */, 2 /* container */, 1L),
+        position(1 /* bitmap */, 2 /* container */, CONTAINER_OFFSET - 1));
+
+    assertThat(bitmap.compress()).as("Bitmap must be RLE encoded").isTrue();
+
+    RoaringPositionBitmap bitmapCopy = roundTripSerialize(bitmap);
+
+    assertThat(bitmapCopy.cardinality()).isEqualTo(bitmap.cardinality());
+    bitmapCopy.forEach(position -> assertThat(bitmap.contains(position)).isTrue());
+    bitmap.forEach(position -> assertThat(bitmapCopy.contains(position)).isTrue());
+  }
+
+  @TestTemplate
+  public void testDeserializeSupportedRoaringExamples() throws IOException {
+    for (String file : SUPPORTED_OFFICIAL_EXAMPLE_FILES) {
+      RoaringPositionBitmap bitmap = readBitmap(file);
+      assertThat(bitmap).isNotNull();
+    }
+  }
+
+  @TestTemplate
+  public void testDeserializeUnsupportedRoaringExample() {
+    // this file contains a value that is larger than the max supported value in our impl
+    assertThatThrownBy(() -> readBitmap("64maphighvals.bin"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("Invalid unsigned key");
+  }
+
+  @TestTemplate
+  public void testUnsupportedPositions() {
+    RoaringPositionBitmap bitmap = new RoaringPositionBitmap();
+
+    assertThatThrownBy(() -> bitmap.add(-1L))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining(
+            "Bitmap supports positions that are >= 0 and <= %s",
+            RoaringPositionBitmap.MAX_POSITION);
+
+    assertThatThrownBy(() -> bitmap.contains(-1L))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining(
+            "Bitmap supports positions that are >= 0 and <= %s",
+            RoaringPositionBitmap.MAX_POSITION);
+
+    assertThatThrownBy(() -> bitmap.add(RoaringPositionBitmap.MAX_POSITION + 1L))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining(
+            "Bitmap supports positions that are >= 0 and <= %s",
+            RoaringPositionBitmap.MAX_POSITION);
+
+    assertThatThrownBy(() -> bitmap.contains(RoaringPositionBitmap.MAX_POSITION + 1L))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining(
+            "Bitmap supports positions that are >= 0 and <= %s",
+            RoaringPositionBitmap.MAX_POSITION);
+  }
+
+  @TestTemplate
+  public void testInvalidSerializationByteOrder() {
+    assertThatThrownBy(() -> RoaringPositionBitmap.deserialize(ByteBuffer.allocate(4)))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("serialization requires little-endian byte order");
+  }
+
+  @TestTemplate
+  public void testRandomSparseBitmap() {
```

Comment:
@RussellSpitzer, here we have 3 tests with random values that are compared against `Set<Long>`.